File: server/src/main/java/dietcourtserver/model/CompositionID.java
package dietcourtserver.model;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Embeddable
public class CompositionID
        implements Serializable {

    @Column(name = "Dish_id")
    private int dishID;

    @Column(name = "Ingredient_id")
    private int ingredientID;

    public CompositionID() {
    }

    public CompositionID(int dishid, int ingrid) {
        this.dishID = dishid;
        this.ingredientID = ingrid;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass())
            return false;

        CompositionID that = (CompositionID) o;
        return Objects.equals(dishID, that.dishID) &&
                Objects.equals(ingredientID, that.ingredientID);
    }

    @Override
    public int hashCode() {
        return Objects.hash(dishID, ingredientID);
    }
}

File: client/app/src/main/java/com/etu/client/data/model/CompositionID.kt
package com.etu.client.data.model

import java.io.Serializable
import java.util.*

class CompositionID(dishID: Int, ingredientID: Int) : Serializable {

    private val dishID = 0

    private val ingredientID = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as CompositionID
        return dishID == that.dishID &&
                ingredientID == that.ingredientID
    }

    override fun hashCode(): Int {
        return Objects.hash(dishID, ingredientID)
    }
}